### ISSUE TITLE
Include duplicates erros

### ISSUE THREAD

**Issue by ERZUSO** [phpactor/phpactor#2437]
I have two files and two questions.

### test.php
```php
<?php

/**
 * @var int $test
 */

if ($test === "gg") {
    echo $test;
}
```

### test2.php
```php
<?php

$test = 1;

if ($test === '1') {
}

include "test.php";
```
```
$ phpactor worse:analyse ./
Analysing files...

 2/2 [============================] 100%

test.php
+-------+----------+----------------------------+
| range | severity | message                    |
+-------+----------+----------------------------+
| 38:43 | ERROR    | Undefined variable "$test" |
| 65:70 | ERROR    | Undefined variable "$test" |
+-------+----------+----------------------------+

test2.php
+-------+----------+----------------------------+
| range | severity | message                    |
+-------+----------+----------------------------+
| 38:43 | ERROR    | Undefined variable "$test" |
| 65:70 | ERROR    | Undefined variable "$test" |
+-------+----------+----------------------------+

4 problems in 0.0380 seconds with 16,613,080b memory
```
1. Is there a way to suppress the 'Undefined variable' diagnoses in `test.php` using PHP DocBlock or another method?
2. Why does the inclusion of `test.php` in `test2.php` result in the 'Undefined variable "$test"' diagnosis for both files?

I tested this on the latest release (2023.09.24) and current master. I used PHP 8.2.

**Comment by dantleech** [comment]
1) not currently --- I guess a typehint outside of a scope/function should count as a "declatation"
2) which version of Phpactor are you using? the inclusion issue may have been fixed in master with #2391 

are you using `worse:analyse` as part of your workflow? or is this just

**Comment by dantleech** [comment]
i can't reprpoduce 1) any more and 2) should be fixed in #2439
